CRM/Case/BAO/Case.php

```diff
@@ -529,6 +529,9 @@ public static function retrieveCaseIdsByContactId($contactID, $includeDeleted =
     if (!$includeDeleted) {
       $query .= " AND ca.is_deleted = 0";
     }
+    if (isset($statusId)) {
+      $query .= " AND ca.status_id = '$statusId'";
+    }
```

Comment:
Quoting is not the same as escaping. Please note how $contactID is escaped in the line below. You will need to move that line above this one, and append $statusId to the $params array, and use a placeholder in the query.